gdal.cmake

```diff
@@ -670,18 +670,13 @@ if (NOT GDAL_ENABLE_MACOSX_FRAMEWORK)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/template/GDALConfig.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/GDALConfig.cmake @ONLY)
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/GDALConfig.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/gdal/)
 
-  # gdal-config utility command generation
-  include(GenerateConfig)
-  if (ENABLE_GNM)
-    set(CONFIG_GNM_ENABLED "yes")
-  else ()
-    set(CONFIG_GNM_ENABLED "no")
-  endif ()
-  get_property(_GDAL_FORMATS GLOBAL PROPERTY GDAL_FORMATS)
-  get_property(_OGR_FORMATS GLOBAL PROPERTY OGR_FORMATS)
-  string(REPLACE ";" " " CONFIG_FORMATS "${_GDAL_FORMATS} ${_OGR_FORMATS}")
-  generate_config(${GDAL_LIB_TARGET_NAME} "gdal_private_link_libraries" ${GDAL_CMAKE_TEMPLATE_PATH}/gdal-config.in
-                  ${PROJECT_BINARY_DIR}/apps/gdal-config)
+  # Generate gdal-config utility command and pkg-config module gdal.pc
+  include(GdalGenerateConfig)
+  gdal_generate_config(
+    TARGET "${GDAL_LIB_TARGET_NAME}"
+    GLOBAL_PROPERTY "gdal_private_link_libraries"
+    GDAL_CONFIG "${PROJECT_BINARY_DIR}/apps/gdal-config"
+    PKG_CONFIG "${CMAKE_CURRENT_BINARY_DIR}/gdal.pc")
   add_custom_target(gdal_config ALL DEPENDS ${PROJECT_BINARY_DIR}/apps/gdal-config)
```

Comment:
Can this target be removed? If you lost the file, you can re-run cmake.

Reply:
> Can this target be removed?

seems fine to remove it

Reply:
Done.